### PR TITLE
feat: track emoji usage, display most frequent emojis on context menu & emoji dialog

### PR DESCRIPTION
**Pull request by genie-magic**
**What this PR does** 📖
Track Emoji Usage, display most frequent emojis on context menu & emoji dialog
**Which issue(s) this PR fixes** 🔨

Fixes # AP-236

**Special notes for reviewers** 🗒️

1. when you not choose emoji, frequently used emoji not appear on context menu & emoji dialog
2. when you click emoji and it is tracked via vuex store and it is showed on context menu & emoji dialog

### CODE REVIEW

File: store/ui/mutations.ts [hunk 1]
```diff
@@ -31,18 +31,30 @@ export default {
...
+      defaultWidth:
+        typeof options.defaultWidth !== 'undefined'
+          ? options.defaultWidth
+          : state.enhancers.defaultWidth,
+      defaultHeight:
+        typeof options.defaultHeight !== 'undefined'
+          ? options.defaultHeight
+          : state.enhancers.defaultHeight,
+      containerWidth:
+        typeof options.containerWidth !== 'undefined'
```
Comment (Jekrimo) [inline store/ui/mutations.ts:54]: Will you see if we have an enum for the string 'undefined' we could throw here for all of these? If not make one real quick if you have time? Definitely a good one for us to have.

Comment (Jekrimo) [reply store/ui/mutations.ts:54]: image

would be here, Maybe common-types.ts ,

**Review by Jekrimo — APPROVED** [review]

**Review by stavares843 — CHANGES_REQUESTED** [review]
the feature itself seems good

but there's a behavior on this branch that doesn't happen on dev


https://user-images.githubusercontent.com/29093946/145851559-19d54408-3376-4813-a172-4d5d86885599.mp4


when you click on an emoji, selects all the emojis, and appears all blue

the same doesn't happen on dev

Chrome, macOS

File: components/tailored/messaging/enhancers/Enhancers.html [hunk 2]
```diff
@@ -6,54 +6,75 @@
...
+      ></div>
       <div slot="emoji-picker" slot-scope="{ emojis, insert, display }">
         <div class="emoji-picker">
           <div class="emoji-picker__search">
-            <InteractablesInput v-model="search" type="dark" size="small" placeholder="Search..." />
+            <InteractablesInput
+              v-model="search"
+              type="dark"
+              size="small"
+              placeholder="Search..."
```
Comment (WanderingHogan) [inline components/tailored/messaging/enhancers/Enhancers.html:47]: Can you put this in the translations

**Review by WanderingHogan — APPROVED** [review]
I think this is good once you have that weird ui thing @stavares843 found, fixed

**Comment by WanderingHogan** [comment]
This needs the highlighting issue fixed, and to be rebased on dev

**Comment by genie-magic** [comment]
> the feature itself seems good
> 
> but there's a behavior on this branch that doesn't happen on dev
> 
>  Screen.Recording.2021-12-13.at.4.31.44.PM.mp4 
> when you click on an emoji, selects all the emojis, and appears all blue
> 
> the same doesn't happen on dev
> 
> Chrome, macOS

@stavares843  @WanderingHogan  , highlight issue is fixed .. plz check.

**Comment by WanderingHogan** [comment]
Looks good to me, @stavares843 if it looks ok to you please merge, we can add the translation updates to https://satellite-im.atlassian.net/browse/AP-277

**Comment by stavares843** [comment]
thanks @WanderingHogan checking 💥

**Review by stavares843 — APPROVED** [review]
neat! 🎉

**Pull request merged**